File: demo/translate/targets/ADemo.cpp
class ADemo : public testing::Test {
public:
    int a;
    Target b;
    Target c = new Target();

    void test1(Target a){
        ADemo::BDemo * d = new ADemo::BDemo();
        d->test2();
    }

    class BDemo {
        public:
            int a;
            std::string b;
            void test2(std::string a){
                printf("hello");
            }
    };
};

File: demo/translate/source/ADemo.java
public class ADemo {
    private int a;
    private Source b,e,f;
    private Source c = new Source();

    public void test1(Source a){
        BDemo d = new BDemo();
        d.test2();
    }

    class BDemo{
        private int a;
        private String b;
        public void test2(String a){
            System.out.println("hello");
        }
    }
}